### PR TITLE
Move credit card agreements to jinja

### PR DESCRIPTION
**Pull request by anselmbradford**
We have redundant references to chosen.js, which should ultimately be removed. As a step toward that, this moves the credit card agreement database page to jinja so we can use the `base.html` template and incorporate CF components.

There are number of opinionated edits in this PR that someone from UX would need to OK. These include:

- Replace custom modal with expandable.
- Create consistent search helper text ("Can't find an issuer?") between agreement landing page and search results page sidebar.
- Create consistent database disclosure and social media share icons at the bottom of both the agreement landing page and search results page sidebar.

## Changes

- Moves agreements app templates to jinja.
- Replace custom modal with expandable and add `single.js` to initialize expandables on /credit-cards/ URL.
- Replace custom PDF download arrow with svg download icon.
- Replaces search results table with cf-table.
- Replace pagination with cf-pagination.
- Export BASE_CLASS from Multiselect for initialization purposes (not yet used, because chosen is still used).

## Testing

1. `./frontend.sh`
2. Visit http://localhost:8000/credit-cards/agreements/ and ensure "Can't find issuer" expandable expands. 
3. Make sure links work on the page.
4. Select chosen.js drop-down of banks to go to the search results page.
5. Perform same checks on search results page.
6. Search for "Citibank" to get a page with pagination at the bottom of the table.
7. Click through pagination pages.

## Screenshots

download-1

download-2

download-3

## Todos

- Chosen.js needs to be replaced with something, though the multiselect does not currently support anything other than 5 options selected.
- Tests are probably broken.

### CODE REVIEW

File: cfgov/agreements/jinja2tags.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+from agreements.models import Agreement, Issuer
+
+def issuer_select(selected=None):
+    issuers = Issuer.objects.all().order_by('name')
+    issuers_filtered = []
+    for issuer in issuers:
+        if Agreement.objects.filter(issuer=issuer).exists():
+            issuers_filtered.append(issuer)
+
+    markup = '<select data-placeholder="Choose an issuer" class="chzn-select" tabindex="2" id="issuer_select">'
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply cfgov/agreements/jinja2tags.py]: False positive.

File: cfgov/agreements/static/agreements/js/jquery-1.9.1.min.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
...ked)}b.extend({clone:function(e,t,n){var r,i,o,a,s,u=b.contains(e.ownerDocument,e);if(b.support.html5Clone||b.isXMLDoc(e)||!mt.test("<"+e.nodeName+">")?o=e.cloneNode(!0):(Dt.innerHTML=e.outerHTML,Dt.removeChild(o=Dt.firstChild)),!(b.support.noCloneEvent&&b.support.noCloneChecked||1!==e.nodeType&&11!==e.nodeType||b.isXMLDoc(e)))for(r=Ot(o),s=Ot(e),a=0;null!=(i=s[a]);++a)r[a]&&Ft(i,r[a]);if(t)if(n)for(s=s||Ot(e),r=r||Ot(o),a=0;null!=(i=s[a]);a++)_t(i,r[a]);else _t(e,o);return r=Ot(o,"script"),r.length>0&&Mt(r,!u&&Ot(e,"script")),r=s=i=null,o},buildFragment:function(e,t,n,r){var i,o,a,s,u,l,c,p=e.length,f=dt(t),d=[],h=0;for(;p>h;h++)if(o=e[h],o||0===o)if("object"===b.type(o))b.merge(d,o.nodeType?[o]:o);else if(wt.test(o)){s=s||f.appendChild(t.createElement("div")),u=(bt.exec(o)||["",""])[1].toLowerCase(),c=At[u]||At._default,s.innerHTML=c[1]+o.replace(vt,"<$1></$2>")+c[2],i=c[0];while(i--)s=s.lastChild;if(!b.support.leadingWhitespace&&yt.test(o)&&d.push(t.createTextNode(yt.exec(o)[0])),!b.support.tbody){o="table"!==u||xt.test(o)?"<table>"!==c[1]||xt.test(o)?0:s:s.firstChild,i=o&&o.childNodes.length;while(i--)b.nodeName(l=o.childNodes[i],"tbody")&&!l.childNodes.length&&o.removeChild(l)
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply cfgov/agreements/static/agreements/js/jquery-1.9.1.min.js]: False positive

File: cfgov/agreements/static/agreements/js/jquery-1.9.1.min.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
...entDefault()}},b.removeEvent=o.removeEventListener?function(e,t,n){e.removeEventListener&&e.removeEventListener(t,n,!1)}:function(e,t,n){var r="on"+t;e.detachEvent&&(typeof e[r]===i&&(e[r]=null),e.detachEvent(r,n))},b.Event=function(e,n){return this instanceof b.Event?(e&&e.type?(this.originalEvent=e,this.type=e.type,this.isDefaultPrevented=e.defaultPrevented||e.returnValue===!1||e.getPreventDefault&&e.getPreventDefault()?it:ot):this.type=e,n&&b.extend(this,n),this.timeStamp=e&&e.timeStamp||b.now(),this[b.expando]=!0,t):new b.Event(e,n)},b.Event.prototype={isDefaultPrevented:ot,isPropagationStopped:ot,isImmediatePropagationStopped:ot,preventDefault:function(){var e=this.originalEvent;this.isDefaultPrevented=it,e&&(e.preventDefault?e.preventDefault():e.returnValue=!1)},stopPropagation:function(){var e=this.originalEvent;this.isPropagationStopped=it,e&&(e.stopPropagation&&e.stopPropagation(),e.cancelBubble=!0)},stopImmediatePropagation:function(){this.isImmediatePropagationStopped=it,this.stopPropagation()}},b.each({mouseenter:"mouseover",mouseleave:"mouseout"},function(e,t){b.event.special[e]={delegateType:t,bindType:t,handle:function(e){var n,r=this,i=e.relatedTarget,o=e.handleObj;
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply cfgov/agreements/static/agreements/js/jquery-1.9.1.min.js]: False positive.

File: cfgov/agreements/jinja2tags.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (anselmbradford) [reply cfgov/agreements/jinja2tags.py]: Will be refactored later when chosen is replaced.

File: cfgov/agreements/jinja2tags.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+def issuer_select(selected=None):
+    issuers = Issuer.objects.all().order_by('name')
+    issuers_filtered = []
+    for issuer in issuers:
+        if Agreement.objects.filter(issuer=issuer).exists():
+            issuers_filtered.append(issuer)
+
+    markup = '<select data-placeholder="Choose an issuer"'
+    markup += ' class="chzn-select" tabindex="2" id="issuer_select">'
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply cfgov/agreements/jinja2tags.py]: False positive.

[the remark this answers is not in the document]
Comment (anselmbradford) [reply cfgov/agreements/jinja2tags.py]: Will be refactored later when chosen is replaced.

**Comment by anselmbradford** [comment]
The bewd tests are failing. I could use guidance on how to migrate those. I tried `t = Template("{{ issuer_select }}")` in place of `t = Template("{% load agreements_extras %}{% issuer_select %} ")`. Is there a good example of tests that went from django to jinja?

**Comment by chosak** [comment]
@anselmbradford I've fixed the backend tests and made a number of other backend-related changes:

1. Removed the now-deprecated `cfgov/templatetags` directory.
1. Gave the Jinja tag a more precise name of `agreements_issuer_select` because all Jinja tags share the same global namespace.
1. Fixed up the Python linting so that `agreements` is treated as first-party code.
1. Moved the `<select>` HTML into a template for consistency anda easier editing later.
1. Optimized the database queries made for that select; now instead of [literally *hundreds* of queries](https://github.com/cfpb/cfgov-refresh/pull/4827/files#diff-4ca68143a21368233d2985bdb7c55e1aL15) (one per issuer) it makes only a single query.

From a backend perspective this PR looks good to me; I know you wanted some further UX review for which I will defer to others.

**Comment by anselmbradford** [comment]
@schaferjh Do you know who owns this page from UX? This brings it into our modern templates, but changes things to do so. See "opinionated edits" and screenshots in the original post for this PR (versus https://www.consumerfinance.gov/credit-cards/agreements/).

**Comment by schaferjh** [comment]
@anselmbradford oh, interesting! When you're back in the office, let's find 5-10 mins for you to demo this for me and then I can make a call on whether I can sign off or if we need to alert the stakeholder. 

Also note, we will be updating the data behind this page soon -- GHE/platform/3293.

**Comment by anselmbradford** [comment]
On hold pending UX approval and adjustments.

**Comment by anselmbradford** [comment]
Rebased with master and incorporated https://github.com/cfpb/cfgov-refresh/pull/4885

**Comment by chosak** [comment]
@anselmbradford I've pushed a commit that fixes the single test that was failing against Python 3. Has this gone through the UX review mentioned above?

**Comment by anselmbradford** [comment]
> @anselmbradford I've pushed a commit that fixes the single test that was failing against Python 3. Has this gone through the UX review mentioned above?

I demoed the changes with Jess and it will need further design/UX input. @schaferjh is this perhaps something that could be done during this reset sprint?

**Comment by schaferjh** [comment]
@anselmbradford please loop in @keelerr and @ascott1 in my stead during my detail. :)

**Comment by ascott1** [comment]
@anselmbradford @keelerr, maybe this is something that @niqjohnson can look at in the coming weeks?

File: cfgov/agreements/jinja2/agreements/_database_disclaimer.html [hunk 5]
```diff
@@ -1,14 +1,72 @@
...
+        Look on the back of the credit card or on your latest
+        monthly statement to find the name of the issuer. The bank,
+        retailer, or other entity whose name appears on the front
+        of the card may not be the actual issuer of the card.
+        You can also check the issuer's website; typically,
+        the bank name is listed at the bottom of the issuer's
+        homepage.
+    </p>
+    <p>
+        <a href="#">
```
Comment (anselmbradford) [inline cfgov/agreements/jinja2/agreements/_database_disclaimer.html]: Should this link anywhere?

File: cfgov/agreements/jinja2/agreements/_database_disclaimer.html [hunk 6]
```diff
@@ -1,14 +1,72 @@
...
+    <p>
+        The drop-down list contains credit card issuers that have
+        submitted credit card agreements as required under
+        Section 204 of the Credit Card Accountability
+        Responsibility and Disclosure Act of 2009 (the CARD Act).
+        If you cannot find a card issuer, it may be that:
+    </p>
+    <ul>
+        <li>
+        The issuer has not yet submitted its agreements
```
Comment (anselmbradford) [inline cfgov/agreements/jinja2/agreements/_database_disclaimer.html]: `<li>` contents could be indented 4 spaces.

File: cfgov/agreements/jinja2/agreements/search.html [hunk 7]
```diff
@@ -14,140 +14,129 @@
...
+                                        type="number"
+                                        min="1"
+                                        max="999"
+                                        pattern="[0-9]*"
+                                        inputmode="numeric"
+                                        value="{{ page.number }}">
+                                of {{ page.paginator.num_pages }}
+                            </label>
+                            <button class="a-btn
+                                            a-btn__link
```
Comment (anselmbradford) [inline cfgov/agreements/jinja2/agreements/search.html]: TRIVIAL: CSS classes should line up.

File: cfgov/agreements/jinja2/agreements/_database_disclaimer.html (hunk 5 above)
Comment (niqjohnson) [reply cfgov/agreements/jinja2/agreements/_database_disclaimer.html]: Fixed in 3a92a97. I originally though jumping back to the top of the page made sense, but I've reconsidered and added an anchor right to the header above the issuer select on each page.

File: cfgov/agreements/jinja2/agreements/_database_disclaimer.html (hunk 6 above)
Comment (niqjohnson) [reply cfgov/agreements/jinja2/agreements/_database_disclaimer.html]: Fixed in 3a92a97

File: cfgov/agreements/jinja2/agreements/search.html (hunk 7 above)
Comment (niqjohnson) [reply cfgov/agreements/jinja2/agreements/search.html]: Fixed in 3a92a97

**Comment by anselmbradford** [comment]
@niqjohnson approve and I'll merge

**Pull request merged**